File: app/src/main/java/com/haishinkit/rtmp/message/RTMPSetChunkSizeMessage.java
package com.haishinkit.rtmp.message;

import java.nio.ByteBuffer;
import com.haishinkit.rtmp.RTMPSocket;

public final class RTMPSetChunkSizeMessage extends RTMPMessage {
    private static final int CAPACITY = 4;

    private int size = 0;

    public RTMPSetChunkSizeMessage() {
        super(Type.CHUNK_SIZE);
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public ByteBuffer encode(RTMPSocket socket) {
        if (socket == null) {
            throw new IllegalArgumentException();
        }
        ByteBuffer buffer = ByteBuffer.allocate(CAPACITY);
        buffer.putInt(getSize());
        return buffer;
    }
}

File: haishinkit/src/main/java/com/haishinkit/rtmp/message/RtmpSetChunkSizeMessage.kt
package com.haishinkit.rtmp.message

import com.haishinkit.rtmp.RtmpConnection
import java.nio.ByteBuffer

/**
 * 5.4.1. Set Chunk Size (1)
 */
internal class RtmpSetChunkSizeMessage : RtmpMessage(TYPE_CHUNK_SIZE) {
    var size: Int = 0
    override var length: Int = CAPACITY

    override fun encode(buffer: ByteBuffer): RtmpMessage {
        buffer.putInt(size)
        return this
    }

    override fun decode(buffer: ByteBuffer): RtmpMessage {
        size = buffer.int
        return this
    }

    override fun execute(connection: RtmpConnection): RtmpMessage {
        connection.socket.chunkSizeC = size
        return this
    }

    companion object {
        private const val CAPACITY = 4
    }
}
